shared-libs/rules-engine/src/provider-wireup.js

```diff
@@ -135,6 +139,48 @@ module.exports = {
   },
 };
 
+let refreshQueue = Promise.resolve();
+const enqueue = callback => {
+  const listeners = [];
+  const eventQueue = [];
+  const emit = evtName => {
+    if (!listeners[evtName]) {
+      return eventQueue.push(evtName);
+    }
+    listeners[evtName].forEach(callback => callback());
+  };
+
+  emit('queued');
+  refreshQueue = refreshQueue.then(() => {
+    emit('running');
+    return callback();
+  });
+
+  refreshQueue.on = (evtName, callback) => {
+    listeners[evtName] = listeners[evtName] || [];
+    listeners[evtName].push(callback);
+    eventQueue.forEach(queuedEvent => queuedEvent === evtName && callback());
+    return refreshQueue;
+  };
+
+  return refreshQueue;
+};
+
+const jumpQueue = result => {
```

Comment:
`jumpQueue` is only ever called when tasks are disabled with an empty array, right? I think it would be easier to read if it were called `disabledResponse` (or something better?) and not take any params.